appservice/workers/transaction_scheduler.go

```diff
@@ -207,7 +207,14 @@ func send(
 	appservice config.ApplicationService,
 	txnID int,
 	transaction []byte,
-) error {
+) (err error) {
+	defer func() {
+		if err != nil {
+			log.WithFields(log.Fields{
+				"appservice": appservice.ID,
+			}).WithError(err).Error("unable to send event")
```

Comment:
I would rather log at the caller's side (the code which calls `send(...)`) than in a `defer` like this.

Reply:
:+1: done